analyzers/tests/SonarAnalyzer.UnitTest/TestCases/SymbolicExecution/Sonar/HashesShouldHaveUnpredictableSalt.cs

```diff
@@ -214,3 +214,15 @@ public void Bar(byte[] passwordBytes)
         }
     }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/7355
+public class AD0001_Repro
+{
+    // This reproduces scenario when S1944 check is not active as a rule, but still present in the exploded graph as an engine check.
+    // It must be part of another rule to reproduce the original problem
+    public void InvalidCastToInterfaceSymbolicExecution_Repro()
+    {
+        int? i = null;
+        int j = (int)i; // Compliant in this rule, would raise S1944 in the old SE
```

Comment:
I'm a bit confused - the file name is `HashesShouldHaveUnpredictableSalt` and the test is for InvalidCastToInterfaceSymbolicExecution.